system/src/system_network_diagnostics.cpp

```diff
@@ -80,9 +80,28 @@ class SignalCache
 #endif
     }
 
+#if HAL_PLATFORM_CELLULAR
+    const CellularGlobalIdentity* getCGI()
+    {
+        system_tick_t m = millis();
+        if (ts_ == 0 || (m - ts_) >= NETWORK_INFO_CACHE_INTERVAL)
+        {
+            cgi_.size = sizeof(cgi_);
+            cgi_.version = CGI_VERSION_LATEST;
+            cellular_global_identity(&cgi_, nullptr);
```

Comment:
Error code is not being checked. `WiFi/Cellular.RSSI()` have special values that indicate an error, here we need to specifically handle errors returned by `cellular_global_identity()` and somehow forward them to the diagnostic sources.

Reply:
Updated.

Reply:
So am I correct in assuming this is the macro that will be used?  Why is it marked `Deprecated` and in the case of error will it just return `nullptr`?
```
// Deprecated
#define CHECK_RETURN(_expr, _val) \
        ({ \
            const auto _ret = _expr; \
            if (_ret < 0) { \
                _LOG_CHECKED_ERROR(_expr, _ret); \
                return _val; \
            } \
            _ret; \
        })
```

Reply:
I marked `CHECK_RETURN()`  as deprecated for a couple of reasons:

- It loses the error code info and thus breaks the convention. If some module needs to perform a conversion between system- and module-specific result codes/return values then it should implement its own local conversion function and/or a helper macro.
- Having a lot of `CHECK_SOMETHING()` macros hits the readability of the code in the long run. It's better to avoid introducing new ones in the global namespace unless absolutely necessary.

With that said I won't mind having globally defined `CHECK_RETURN()` and `CHECK_GOTO()`, but they should be implemented in a way that preserves the error code info.